### PR TITLE
delete words when longpress key

### PR DESCRIPTION
**Pull request by suntabu**

### CODE REVIEW

**Comment by menny** [comment]
That makes sense.
Can you write a unit-test for that?

**Comment by suntabu** [comment]
@menny I've added unit test for it, if something wrong, please let me know. 😄

File: app/src/main/res/xml/qwerty.xml [hunk 1]
```diff
@@ -54,6 +55,6 @@
         <Key android:codes="98" android:keyLabel="b" android:popupCharacters="β"/>
         <Key android:codes="110" android:keyLabel="n" android:popupCharacters="ñńν"/>
         <Key android:codes="109" android:keyLabel="m"  android:popupCharacters="μ"/>
-        <Key android:keyWidth="15%p" android:codes="-5" android:keyEdgeFlags="right" android:isRepeatable="true"/>
+        <Key android:keyWidth="15%p" android:codes="-5" ask:longPressCode="-7" android:keyEdgeFlags="right" android:isRepeatable="true"/>
     </Row>
```
Comment (menny) [inline app/src/main/res/xml/qwerty.xml]: Please remove the  `longPressCode`, I think I'll add code later that dynamically modifies the delete key.

File: app/src/test/java/com/anysoftkeyboard/keyboards/views/AnyKeyboardViewBaseTest.java [hunk 2]
```diff
@@ -152,6 +153,18 @@ public void testWithLongPressOutputRegularPressKeyPressState() {
...
+        final AnyKeyboard.AnyKey key = findKey(KeyCodes.DELETE);
+        key.longPressCode = -7;
+
+        mUnderTest.onLongPress(mEnglishKeyboard.getKeyboardAddOn(), key, false, mMockPointerTrack);
+
+        Mockito.verify(mMockKeyboardListener).onKey(Mockito.eq((int) KeyCodes.DELETE_WORD), Mockito.same(key), Mockito.eq(0), Mockito.any(int[].class), Mockito.eq(true));
+        Mockito.verify(mMockKeyboardListener, Mockito.never()).onKey(Mockito.eq(key.getPrimaryCode()), Mockito.any(Keyboard.Key.class), Mockito.anyInt(), Mockito.any(int[].class), Mockito.anyBoolean());
+    }
+
+
```
Comment (menny) [inline app/src/test/java/com/anysoftkeyboard/keyboards/views/AnyKeyboardViewBaseTest.java]: There is an extra line here

File: app/src/test/java/com/anysoftkeyboard/keyboards/views/AnyKeyboardViewBaseTest.java [hunk 3]
```diff
@@ -152,6 +153,18 @@ public void testWithLongPressOutputRegularPressKeyPressState() {
         Assert.assertArrayEquals(provider.KEY_STATE_NORMAL, key.getCurrentDrawableState(provider));
     }
 
+    @Test
```
Comment (menny) [inline app/src/test/java/com/anysoftkeyboard/keyboards/views/AnyKeyboardViewBaseTest.java]: Also add a test that verify that long press is repeated.
I think that's possible using `ViewTestUtils.navigateFromTo

**Comment by menny** [comment]
This is great. Just a few changes

**Comment by menny** [comment]
@suntabu I'll pull your commits, fix the little things left and submit another PR

**Comment by menny** [comment]
Forget about my comment above. I created a PR on your fork: https://github.com/suntabu/AnySoftKeyboard/pull/2
Please, merge that PR and I'll be able to merge this one.

**Comment by suntabu** [comment]
@menny  merged!

**Pull request merged**